dev/nix/flake.nix

```diff
@@ -255,7 +255,7 @@
 
             controller-tools = pkgs.buildGoModule rec {
               pname = "controller-tools";
-              version = "0.4.1"; # upgrade this in the Makefile if upgraded here
+              version = "0.14.0"; # upgrade this in the Makefile if upgraded here
```

Comment:
Anyone know why it built without having to change sha256 or vendorHash?

Reply:
Good question; this is beyond my knowledge of Nix, cc @isubasinghe

Reply:
Did you run `nix develop`, this package isn't part of the build only the shell.
```console
error: hash mismatch in fixed-output derivation '/nix/store/ajavbdx0j71cp0fp509z84anmgmc29mv-source.drv':
         specified: sha256-NQlSP9hRLXr+iZo0OeyF1MQs3PourQZN0I0v4Wv5dkE=
            got:    sha256-G0jBQ12cpjfWGhXYppV9dB2n68bExi6ME9QbxXsUWvw=
```

Reply:
Any idea why it doesn't throw the error for me?

```
(devenv) @blkperl ➜ ~/go/src/github.com/argoproj/argo-workflows (controller-gen) $ nix develop --extra-experimental-features nix-command --extra-experimental-features flakes ./dev/nix/ 
wait-for minio
deployment.apps/minio condition met
port-forward minio 9000
Forwarding from 127.0.0.1:9000 -> 9000
Forwarding from [::1]:9000 -> 9000
port-forward minio 9001
Forwarding from 127.0.0.1:9001 -> 9001
Forwarding from [::1]:9001 -> 9001
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.54s.
(devenv) @blkperl ➜ ~/go/src/github.com/argoproj/argo-workflows (controller-gen) $ 
```

Reply:
Weird, could it be a a cache thing perhaps?